pages/PreviewPage/[...picID].js

```diff
@@ -94,49 +120,57 @@ export default function PreviewPage() {
 
   async function handleOnClick() {
     setIsLoadingState(true);
-    await trigger({ searchID, picSRC, picSRCSlug });
+    await trigger({ picID, picSRC, picSRCSlug });
     setIsLoadingState(false);
-    router.push(`/Variations/${searchID[0]}`);
+    router.push(`/Variations/${picID[0]}`);
   }
 
   async function favoriteImage() {
-    const check = favPictures.some((picture) => picture.picID === searchID[1]);
+    const check = favPictures.some((picture) => picture.picID === picID[1]);
+
     if (check === false) {
-      const cloudinaryData = await safeFavToCloud(shirts.picSRC);
-      safeFavToMongoDB(cloudinaryData, `${searchID[1]}`);
+      const cloudinaryData = await safeFavToCloud(picSRC);
+      await safeFavToMongoDB(cloudinaryData, `${picID[1]}`);
```

Comment:
```suggestion
      await saveFavToMongoDB(cloudinaryData, `${picID[1]}`);
```
As mentioned before, I would choose another function name here.